rs/tests/system_tests.bzl

```diff
@@ -49,6 +50,16 @@ def _run_system_test(ctx):
             fi
             unset RUN_SCRIPT_ICOS_IMAGES RUN_SCRIPT_UPLOAD_SYSTEST_DEP # clean up the env for the test
 
+            if [ -z "${{ENV_DEPS__HOSTOS_UPDATE_IMG_VERSION:-}}" ]; then
```

Comment:
I'm a bit uneasy about this extra logic in the general system_test macro for the specific nested tests. Ideally this should be moved to the call sites under rs/tests/nested. However I'm fine if we do this in a follow up PR since I currently don't have enough time to write a good suggestion for an alternative.

Reply:
This is here so we can re-use the image download logic above, right? For now could we make this a little more explicit with an extra `RUN_SCRIPT_XXX` variable? That way we can depend on the presence of the new variable, rather than the absence of the others.